test/warp.jl

```diff
@@ -387,4 +387,40 @@ ref_img_pyramid_grid = Float64[
             @test nearlysame(round.(Float64.(imgrq_cntr[axes(imgrq_cntr)...]), digits=3), round.(ref_img_pyramid_quad, digits=3))
         end
     end
+
+    @testset "imrotate" begin
+        # imrotate is a wrapper, hence only need to guarantee the interface works as expected
+
+        test_types = (Float32, Float64, N0f8, N0f16)
+        graybar = repeat(range(0,stop=1,length=100),1,100)
+        @testset "interface" begin
+            for T in test_types
+                img = Gray{T}.(graybar)
+                @test_nowarn imrotate(img, 45)
+                @test_nowarn imrotate(img, 45, Constant())
+                @test_nowarn imrotate(img, 45, Linear())
+                @test_nowarn imrotate(img, 45, axes(img))
+                @test_nowarn imrotate(img, 45, axes(img), Constant())
+            end
+        end
+
+        @testset "numerical" begin
+            repeats = 50
+            for T in test_types[3:4]
+                img = Gray{T}.(graybar)
+                for i in 1:repeats
+                    p = 2pi*randn()
+                    @test imrotate(img,p) == imrotate(img,p+2pi)
+                end
+            end
+            # FIXME: Floats are not accurate enough due to RotMatrix is not accurate
+            for T in test_types[1:2]
+                img = Gray{T}.(graybar)
+                for i in 1:repeats
+                    p = 2pi*randn()
+                    @test_broken imrotate(img,p) ≈ imrotate(img,p)
```

Comment:
how can this test test possibly be broken? Is this function non-deterministic?

Reply:
Sorry for the wrong comment. It's the behavior of `NaN` since the default filling value is `NaN` is supported. `Float` supports `NaN` and `N0f8` doesn't.
Change to `isequal` shall be fine.

```julia
julia> NaN == NaN
false
julia> isequal(NaN, NaN)
true
```

Reply:
oh, right. good point.